### PR TITLE
16 pythoncalls

### PR DESCRIPTION
**Pull request by aradhakrishnanGFDL**
addresses #16 

summary of changes

1- Capability to use catalog builder in a cleaner way from notebook and .py 
2- Updated examples in scripts/
3- Added new pytests cases to test this functionality 
4- Added additional CI workflow to test a locally built conda package (this may need a different approach later, but works well for me) 

note - fre-cli needs an update to use create_catalog rather than main
see example here https://github.com/aradhakrishnanGFDL/forkCatalogBuilder-/blob/16-pythoncalls/catalogbuilder/scripts/gen_intake_gfdl_runner.py#L10

### CODE REVIEW

**Comment by aradhakrishnanGFDL** [comment]
This will allow analysis-scripts repository to have a nicer interface to catalogbuilder for testing

**Comment by aradhakrishnanGFDL** [comment]
@ceblanton context - this helps us change this https://github.com/NOAA-GFDL/analysis-scripts/blob/main/tests/test_freanalysis_clouds.py#L49 and do something like https://github.com/aradhakrishnanGFDL/forkCatalogBuilder-/blob/16-pythoncalls/catalogbuilder/scripts/gen_intake_gfdl_runner.py#L10

@menzel-gfdl keeping you in the loop.

File: .github/workflows/conda-pkg-extra-tests.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        conda build .
+    - name: Run additional utilities as tests
+      run: |
+        conda create --name catalogbuildertest 
+        conda install -n catalogbuildertest catalogbuilder --use-local
+        /opt/conda/envs/catalogbuildertest/bin/pytest catalogbuilder/tests/test_create_catalog.py
+        #/opt/conda/envs/catalogbuildertest/bin/python catalogbuilder/tests/make_sample_data.py
+        #/opt/conda/envs/catalogbuildertest/bin/python catalogbuilder/scripts/gen_intake_gfdl_runner.py
+        #we will save the output from following alone as manifest
+        #/opt/conda/envs/catalogbuildertest/bin/python catalogbuilder/scripts/gen_intake_gfdl_runner_config.py
```
Comment (ceblanton) [inline .github/workflows/conda-pkg-extra-tests.yml:29]: Maybe remove these commented out tests?

**Review by ceblanton — APPROVED** [review]
Looks good

**Pull request merged**